src/main/scala/org/scalajs/tools/tsimporter/parser/TSDefParser.scala

```diff
@@ -54,13 +50,16 @@ class TSDefParser extends StdTokenParsers with ImplicitConversions {
     phrase(ambientDeclarations)(new lexical.Scanner(input))
 
   lazy val ambientDeclarations: Parser[List[DeclTree]] =
-    rep(ambientDeclaration)
+    rep(ambientDeclaration.map(Some(_)) | ignoredAmbientDeclaration).map(_.flatMap(_.toList))
 
   lazy val ambientDeclaration: Parser[DeclTree] = (
       opt("declare") ~> opt("export") ~> moduleElementDecl1
     | opt("export") ~> opt("declare") ~> moduleElementDecl1
   )
 
+  lazy val ignoredAmbientDeclaration: Parser[Option[DeclTree]] =
```

Comment:
I would prefer to change the type of `ambientDeclaration` to return `Parser[Option[DeclTree]]` rather than splitting the grammar into `ambientDeclaration` and `ignoredAmbientDeclaration`. The parser structure should more or less follow the grammar of TypeScript, and not be affected by what is or is not ignored in this particular iteration of the importer.

Reply:
That makes sense. I will make the necessary changes and update the PR.

Reply:
This should be addressed in 32042bd, @sjrd.